iceoryx_posh/test/moduletests/test_capro_service.cpp

```diff
@@ -28,166 +28,697 @@ using namespace iox::capro;
+    ServiceDescription description2 = ServiceDescription(serviceID2, iox::capro::AnyEvent, iox::capro::AnyInstance);
+
+    EXPECT_FALSE(iox::capro::serviceMatch(description1, description2));
+}
+
+TEST_F(ServiceDescription_test, ClassHashIndexOperatorReturnsTheValueInTheCorrespondingIndex)
+{
+    ServiceDescription::ClassHash testHash{1U, 2U, 3U, 4U};
+
+    EXPECT_THAT(testHash[0], Eq(1U));
+    EXPECT_THAT(testHash[1], Eq(2U));
+    EXPECT_THAT(testHash[2], Eq(3U));
+    EXPECT_THAT(testHash[3], Eq(4U));
+}
+
+TEST_F(ServiceDescription_test, ComparingTwoUnequalClassHashWithEqualityOperatorReturnsFalse)
+{
+    ServiceDescription::ClassHash testHash1{1U, 2U, 3U, 4U};
+    ServiceDescription::ClassHash testHash2{5U, 6U, 7U, 8U};
+
+    EXPECT_FALSE(testHash1 == testHash2);
+}
+
+TEST_F(ServiceDescription_test, ComparingTwoUnequalClassHashWithEqualityOperatorReturnsTrue)
+{
+    ServiceDescription::ClassHash testHash1{1U, 2U, 3U, 4U};
+    ServiceDescription::ClassHash testHash2{1U, 2U, 3U, 4U};
+
+    EXPECT_TRUE(testHash1 == testHash2);
+}
+
+TEST_F(ServiceDescription_test, ComparingTwoUnequalClassHashWithNotEqualOperatorReturnsTrue)
+{
+    ServiceDescription::ClassHash testHash1{1U, 2U, 3U, 4U};
+    ServiceDescription::ClassHash testHash2{5U, 6U, 7U, 8U};
+
+    EXPECT_TRUE(testHash1 != testHash2);
+}
+
+TEST_F(ServiceDescription_test, ComparingTwoEqualClassHashWithNotEqualOperatorReturnsFalse)
+{
+    ServiceDescription::ClassHash testHash1{1U, 2U, 3U, 4U};
+    ServiceDescription::ClassHash testHash2{1U, 2U, 3U, 4U};
+
+    EXPECT_FALSE(testHash1 != testHash2);
+}
+
+TEST_F(ServiceDescription_test, IsInternalMethodReturnsTrueWhenTheScopeIsSetToInternal)
+{
+    uint16_t testServiceID = 1U;
+    uint16_t testEventID = 2U;
+    uint16_t testInstanceID = 3U;
+    ServiceDescription serviceDescription1 = ServiceDescription(testServiceID, testEventID, testInstanceID);
+
+    serviceDescription1.setInternal();
+
+    EXPECT_TRUE(serviceDescription1.isInternal());
+}
+
+TEST_F(ServiceDescription_test, GetScopeMethodReturnsTheCorrespondingValueOfScope)
+{
+    uint16_t testServiceID = 1U;
+    uint16_t testEventID = 2U;
+    uint16_t testInstanceID = 3U;
+    ServiceDescription serviceDescription1 = ServiceDescription(testServiceID, testEventID, testInstanceID);
+
+    serviceDescription1.setInternal();
+
+    EXPECT_EQ(serviceDescription1.getScope(), Scope::INTERNAL);
+}
+
+TEST_F(ServiceDescription_test,
+       ServiceDescriptionIsInvalidWhen_m_hasServiceOnlyDescriptionIsTrueAndServiceStringIsInvalid)
+{
+    uint16_t testServiceID = iox::capro::AnyService;
+    uint16_t testEventID = iox::capro::AnyEvent;
+    uint16_t testInstanceID = iox::capro::AnyInstance;
+    testService = iox::capro::InvalidIDString;
+    testEvent = "validevent";
+    testInstance = "validinstance";
+    ServiceDescription::ClassHash testHash{1U, 2U, 3U, 4U};
+    auto serialObj = iox::cxx::Serialization::create(testService,
```

Comment:
There no need to use the Serialization here, just use the normal Ctors of the ServiceDescription, they set for you the bool then rightly. The same with the tests below.
Generally the purpose of the Serialization is not to be an alternative Constructor. It is intended to send/receive the ServiceDescription over communication protocols which transfers strings like the MessageQueue.